### PR TITLE
9387 pypi push

### PR DESCRIPTION
**Pull request by gplsteph**
Added a github workflow to package to pip on new releases
- Retrieve the release number from the tag.
- Try to publish on test-pypi and only then on official pypi.
- Changed shotgun-api dependencies to something which works with pypi.

Example: https://test.pypi.org/manage/project/sg-otio/release/1.0.0rc1/

### CODE REVIEW

File: .github/workflows/publish-to-pypi.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+#  push:
+#    tags:
+#    - "*"
+
+jobs:
+  build-n-publish:
+    name: Build and publish Python 🐍 distributions 📦 to PyPI and TestPyPI
+    strategy:
+      matrix:
+        python-version: ['3.10']  # Not sure if we should build for Python 2?
```
Comment (ksallee) [inline .github/workflows/publish-to-pypi.yaml:19]: I think we should? Generally the more versions of python you support the better?

File: setup.py [hunk 2]
```diff
@@ -28,7 +68,7 @@
     },
     install_requires=[
         "OpenTimelineIO >= 0.12.0",
-        "shotgun_api3 @ git+https://github.com/shotgunsoftware/python-api.git",
+        "shotgun-api3 >= 3.3.3",
```
Comment (ksallee) [inline setup.py:71]: How does that work, since shotgun-api3 is not available on Pypi, only github?

File: setup.py [hunk 3]
```diff
@@ -54,6 +94,8 @@
         "Programming Language :: Python :: 3",
         "Programming Language :: Python :: 3.7",
         "Programming Language :: Python :: 3.8",
+        "Programming Language :: Python :: 3.9",
```
Comment (ksallee) [inline setup.py:97]: aren't all versions listed here supposed to go into the matrix for the pypi job?

**Review by ksallee — APPROVED** [review]
LGTM, although not sure about shotgun-api3 without github URL, and about which versions of python to put into the publish job.

File: setup.py (hunk 3 above)
Comment (gplsteph) [reply setup.py:97]: No idea, and not sure how to name wheels with the right py version

File: setup.py (hunk 2 above)
Comment (ksallee) [reply setup.py:71]: Why not add the `>=` and keep the `@`? Isn't it possible?
If some dependency needs to be installed, it needs to be in the installation README.

Comment (ksallee) [reply setup.py:71]: And how is the CI machine installing this if it has no access to it?

Comment (gplsteph) [reply setup.py:71]: not possible, should be added to the README, yes. I'll add a polish ticket for all of that.

File: .github/workflows/ci.yaml [hunk 4]
```diff
@@ -54,6 +54,7 @@ jobs:
       - name: Install dependencies
         run: |
           python -m pip install --upgrade pip
+          pip install six
```
Comment (ksallee) [inline .github/workflows/ci.yaml:57]: How come you need to install six when it is in the `install_requires`?

File: setup.py (hunk 2 above)
Comment (gplsteph) [reply setup.py:72]: I added `pip install git+https://github.com/shotgunsoftware/python-api.git` to the CI

File: README.md [hunk 5]
```diff
@@ -20,6 +20,11 @@ contributions.
 Installation
 ------------
 
+### SG Python API
+
+The SG Python API is not available on PyPi and must be manually installed
+from github, e.g. `pip install git+https://github.com/shotgunsoftware/python-api.git`
```
Comment (ksallee) [inline README.md:26]: That won't update the package if you have an older version installed. I guess the user gets an error when installing and it's up to him to understand it?

**Comment by gplsteph** [comment]
Tests are passing now, and I did a small tweak to the README. Please have another look.

**Review by ksallee — APPROVED** [review]
LGTM

**Pull request merged**